### PR TITLE
bugfix: fix the full table scan issue with 'setDate' condition in Oracle 10g

### PR DESCRIPTION
**Pull request by slievrly**
- [ ] I have registered the PR [changes](../changes).

### Ⅰ. Describe what this PR did

fix the full table scan issue with 'setDate' condition in Oracle 10g

### CODE REVIEW

**Review by a364176773 — COMMENTED** [review]
这里我抛个问题出来
1.超过7天的undolog是否需要只删除status=1的情况? 因为这个时间是可以由用户改的,很可能改成1天,那么被脏写的数据的undolog会被删除,到时候可能又会让社区背锅
2.目前rt侧的metrics没有,有必要把那个client metrics合并一下,因为server侧的metrics还是比较总体的,client有了后,针对这块像业务方就可以自行配置相关的监控告警,不需要由基建运维人员去做相关的转达了

**Review by a364176773 — APPROVED** [review]
LGTM 1.x也合并下这个pr吧,如果考虑到未来可能要加上status做删除,不妨将status和datetime做个组合索引吧

File: script/client/at/db/mysql.sql [hunk 1]
```diff
@@ -9,6 +9,7 @@ CREATE TABLE IF NOT EXISTS `undo_log`
     `log_created`   DATETIME(6)  NOT NULL COMMENT 'create datetime',
     `log_modified`  DATETIME(6)  NOT NULL COMMENT 'modify datetime',
     UNIQUE KEY `ux_undo_log` (`xid`, `branch_id`)
```
Comment (slievrly) [inline script/client/at/db/mysql.sql:11]: Considering that a service is mapped to different seata-server clusters by different downstream service invocations may result in branchId conflicts. Leave xid+branchId as the unique key constraint.

---

考虑一个服务被不同下游服务调用映射不同的seata-server集群，可能导致branchId冲突。保留xid+branchId作为唯一键约束。

Comment (slievrly) [reply script/client/at/db/mysql.sql:11]: done.

**Pull request merged**